docker/dockerfiles/Dockerfile-dev-ubuntu

```diff
@@ -1,10 +1,34 @@
-FROM starrocks/dev-toolchain-ubuntu:main
+FROM ubuntu:22.04 as base
 
+# Install common libraries and tools that are needed for dev environment
+RUN apt-get update -y && \
+    apt-get install --no-install-recommends -y \
+    automake binutils-dev bison byacc ccache flex libiberty-dev libtool maven zip python3 python-is-python3 make cmake gcc g++ default-jdk git patch lld bzip2 \
+    wget unzip curl vim tree net-tools openssh-client && \
+    rm -rf /var/lib/apt/lists/*
+
+ENV JAVA_HOME=/lib/jvm/default-java
 ENV STARROCKS_THIRDPARTY=/opt/starrocks/thirdparty
 ENV STARROCKS_LINKER=lld
 
 WORKDIR /root
 
-# copy installed thirdparty to the image
-COPY thirdparty/installed /opt/starrocks/thirdparty/installed
+FROM base as builder
+COPY . ./starrocks
+
+# build third-party dependencies
+RUN mkdir -p $STARROCKS_THIRDPARTY/installed && cd starrocks/thirdparty &&\
+     ./build-thirdparty.sh && cp -r installed $STARROCKS_THIRDPARTY/
```

Comment:
how about removing ./starrocks/thirdparty, because the compile intermediate data will occupy a lot of storage.

Reply:
That's just an intermediate docker image. 
The eventually image is based on https://github.com/StarRocks/starrocks/pull/16890/files#diff-e27129ed5bdba5985005b98f6cd71dbcc0d7181ce8cf4d46baec9f643326e9e9R1

Reply:
Yes, but a smaller intermediate image is better without other flaws.

Reply:
FYI,
Here are the sizes of each different variant. You can see the intermediate image are very big but the final image is small.

```
ubuntu@ip-172-31-12-223:~/git/starrocks$ docker image ls
REPOSITORY           TAG       IMAGE ID       CREATED         SIZE
dengliu/sr-dev-env   latest    4a6fdc7bc47a   6 hours ago     11GB <=== final image
<none>               <none>    6e036b32f305   9 hours ago     52.6GB <=== intermediate image without removing starrocks/ dir
<none>               <none>    cff8e8a778a6   19 hours ago    29.3GB <=== intermediate image with starrocks/ dir removed 

```

Reply:
Yes if you add `rm -rf starrocks/thirdparty` after coping it will make the intermediate image much smaller

Reply:
I think it's better to add  `docker build --rm=true` to ensure intermediate images get removed after build succeed.

Reply:
added this to the instruction:
https://github.com/StarRocks/starrocks/pull/16890/files#diff-e27129ed5bdba5985005b98f6cd71dbcc0d7181ce8cf4d46baec9f643326e9e9R4